client/my-sites/stats/site.jsx

```diff
@@ -267,23 +265,75 @@ class StatsSite extends Component {
 					</div>
 				</div>
 				<JetpackColophon />
+			</>
+		);
+	}
+
+	renderEnableStatsModule() {
+		const { slug, siteId, path } = this.props;
+		return (
+			<EmptyContent
+				illustration="/calypso/images/illustrations/illustration-404.svg"
+				title={ translate( 'Looking for Stats?' ) }
+				line={ translate(
+					'Enable Site Stats below or visit {{a}}Traffic > Site Stats{{/a}} for more options',
+					{
+						components: {
+							a: <a href={ `/marketing/traffic/${ slug }` } />,
+						},
+					}
+				) }
+				action={ translate( 'Enable Site Stats' ) }
+				actionCallback={ () => this.props.enableJetpackStatsModule( siteId, path ) }
+			/>
+		);
+	}
+
+	render() {
+		const { isJetpack, siteId, showEnableStatsModule } = this.props;
+
+		const { period } = this.props.period;
+
+		return (
+			<Main wideLayout={ true }>
+				<QueryKeyringConnections />
+				{ isJetpack && siteId && <QueryJetpackModules siteId={ siteId } /> }
+				{ siteId && <QuerySiteKeyrings siteId={ siteId } /> }
+				<DocumentHead title={ translate( 'Stats and Insights' ) } />
+				<PageViewTracker
+					path={ `/stats/${ period }/:site` }
+					title={ `Stats > ${ titlecase( period ) }` }
+				/>
+				{ showEnableStatsModule ? this.renderEnableStatsModule() : this.renderStats() }
 			</Main>
 		);
 	}
 }
+const enableJetpackStatsModule = ( siteId, path ) =>
+	withAnalytics(
+		recordTracksEvent( 'calypso_jetpack_module_toggle', {
+			module: 'stats',
+			path,
+			toggled: 'on',
+		} ),
+		activateModule( siteId, 'stats' )
+	);
 
 export default connect(
 	state => {
 		const siteId = getSelectedSiteId( state );
 		const isJetpack = isJetpackSite( state, siteId );
-
+		const isStatsModuleActive = isJetpackModuleActive( state, siteId, 'stats' );
```

Comment:
Since we are using `isStatsModuleActive` only once, and it will only be done for Jetpack sites, should we move it directly below to the `showEnableStatsModule` line? That way this selector will be called only if the site is a Jetpack site.

Reply:
Selectors should be pretty cheap to run, but I can move this inline if you want you'd prefer the optimization. Readability takes a minor hit though, so I'd still take individual declarations on a longer check. Updated in dae8a42